chimney/src/main/scala/io/scalaland/chimney/internal/TransformerMacros.scala

```diff
@@ -108,7 +110,32 @@ trait TransformerMacros {
 
   def expandTargetWrappedInOption(srcPrefixTree: Tree, config: Config)(From: Type,
                                                                        To: Type): Either[Seq[DerivationError], Tree] = {
-    Right(q"_root_.scala.Option[$From]($srcPrefixTree)")
+    val toInnerT = To.typeArgs.head
+    if (To <:< noneTpe) {
+      Left {
+        Seq(NotSupportedDerivation(From.typeSymbol.fullName.toString, To.typeSymbol.fullName.toString))
+      }
+    } else if (toInnerT =:= From) {
+      Right(q"_root_.scala.Option[$From]($srcPrefixTree)")
+    } else {
+      expandTransformerTree(q"$srcPrefixTree", config.rec)(From, toInnerT).mapRight { innerTransformer =>
+        q"_root_.scala.Option[$toInnerT]($innerTransformer)"
+      }
+    }
+  }
+
+  def expandSourceWrappedInOption(srcPrefixTree: Tree, config: Config)(From: Type,
+                                                                       To: Type): Either[Seq[DerivationError], Tree] = {
+    val fromInnerT = From.typeArgs.head
+    if (From <:< noneTpe) {
```

Comment:
That’s nice. Could you please add test case that checks this behavior?

Reply:
Sure of course